### PR TITLE
Rework settings

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/settings/SettingsLoader.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.repository.settings;
+
+import java.util.Map;
+
+public interface SettingsLoader {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/settings/SettingsLoader.java]: CRITICAL Annotate the "SettingsLoader" interface with the @FunctionalInterface annotation [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClientProvider.java [hunk 2]
```diff
@@ -31,15 +31,15 @@
...
   static final String READ_TIMEOUT_SEC_PROPERTY = "sonar.ws.timeout";
   static final int DEFAULT_READ_TIMEOUT_SEC = 60;
 
-  private BatchWsClient wsClient;
+  private ScannerWsClient wsClient;
 
-  public synchronized BatchWsClient provide(final GlobalProperties settings, final EnvironmentInformation env) {
+  public synchronized ScannerWsClient provide(final GlobalProperties settings, final EnvironmentInformation env, GlobalMode globalMode) {
     if (wsClient == null) {
       String url = defaultIfBlank(settings.property("sonar.host.url"), CoreProperties.SERVER_BASE_URL_DEFAULT_VALUE);
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClientProvider.java]: MINOR Remove this use of "SERVER_BASE_URL_DEFAULT_VALUE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 20 issues
* BLOCKER 1 blocker
* CRITICAL 6 critical
* MAJOR 3 major
* MINOR 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. BLOCKER [DefaultAnalysisMode.java#L38](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java#L38): "mediumTestMode" is the name of a field in "AbstractAnalysisMode". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2387)
1. CRITICAL [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)
1. CRITICAL [ScannerExtensionDictionnary.java#L234](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L234): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerExtensionDictionnary.java#L257](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L257): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [ScannerPluginJarExploder.java#L65](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginJarExploder.java#L65): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerPluginPredicate.java#L74](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginPredicate.java#L74): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2638)
1. MAJOR [DefaultAnalysisMode.java#L45](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java#L45): Add the "@Override" annotation above this method signature [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Provide multiple methods instead of using "sort" to determine which action to take. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)
1. MAJOR [ScannerWsClient.java#L119](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClient.java#L119): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. MINOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/e8d432f58f81f19d841637a87de351e720adaa9a/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/ServerSideProjectDataLoader.java [hunk 3]
```diff
@@ -19,7 +19,6 @@
  */
 package org.sonar.scanner.repository;
 
-
-public interface ProjectRepositoriesLoader {
-  ProjectRepositories load(String projectKeyWithBranch, boolean issuesMode);
+public interface ServerSideProjectDataLoader {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/ServerSideProjectDataLoader.java]: CRITICAL Annotate the "ServerSideProjectDataLoader" interface with the @FunctionalInterface annotation [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java [hunk 4]
```diff
@@ -36,17 +35,12 @@
   private static final Logger LOG = LoggerFactory.getLogger(DefaultAnalysisMode.class);
   private static final String KEY_SCAN_ALL = "sonar.scanAllFiles";
 
-  private boolean mediumTestMode;
   private boolean scanAllFiles;
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java]: BLOCKER "mediumTestMode" is the name of a field in "AbstractAnalysisMode". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2387)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java [hunk 5]
```diff
@@ -36,17 +35,12 @@
...
   public DefaultAnalysisMode(GlobalProperties globalProps, AnalysisProperties props) {
     init(globalProps.properties(), props.properties());
   }
 
-  public boolean isMediumTest() {
-    return mediumTestMode;
-  }
-
   public boolean scanAllFiles() {
     return scanAllFiles;
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/analysis/DefaultAnalysisMode.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)

**Comment by SonarTech** [comment]
SonarQube analysis reported 21 issues
* BLOCKER 1 blocker
* CRITICAL 7 critical
* MAJOR 3 major
* MINOR 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)
1. CRITICAL [ScannerExtensionDictionnary.java#L234](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L234): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerExtensionDictionnary.java#L257](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L257): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [ScannerPluginJarExploder.java#L65](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginJarExploder.java#L65): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerPluginPredicate.java#L74](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginPredicate.java#L74): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2638)
1. MAJOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Provide multiple methods instead of using "sort" to determine which action to take. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)
1. MAJOR [ScannerWsClient.java#L119](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClient.java#L119): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. MINOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L98](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L98): Remove this use of "Sensor"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* CRITICAL 7 critical
* MAJOR 2 major
* MINOR 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)
1. CRITICAL [ScannerExtensionDictionnary.java#L234](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L234): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerExtensionDictionnary.java#L257](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L257): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [ScannerPluginJarExploder.java#L65](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginJarExploder.java#L65): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerPluginPredicate.java#L74](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginPredicate.java#L74): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2638)
1. MAJOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Provide multiple methods instead of using "sort" to determine which action to take. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)
1. MAJOR [ScannerWsClient.java#L119](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClient.java#L119): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. MINOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L98](https://github.com/SonarSource/sonarqube/blob/dd6de28927116a4dbd6b659a4dea9d55a1673c78/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L98): Remove this use of "Sensor"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 19 issues
* CRITICAL 7 critical
* MAJOR 2 major
* MINOR 10 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)
1. CRITICAL [ScannerExtensionDictionnary.java#L234](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L234): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerExtensionDictionnary.java#L257](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L257): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [ScannerPluginJarExploder.java#L65](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginJarExploder.java#L65): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)
1. CRITICAL [ScannerPluginPredicate.java#L74](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginPredicate.java#L74): Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2638)
1. MAJOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Provide multiple methods instead of using "sort" to determine which action to take. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)
1. MAJOR [ScannerWsClient.java#L119](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClient.java#L119): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. MINOR [ScannerExtensionDictionnary.java#L73](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L73): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L95](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L95): Remove this use of "Project"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [ScannerExtensionDictionnary.java#L98](https://github.com/SonarSource/sonarqube/blob/c992b8feb6ed86e4abf9f66af94b8f6778f54751/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerExtensionDictionnary.java#L98): Remove this use of "Sensor"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by henryju** [comment]
I'm closing this P/R since we'll not change project/module settings loading

**Pull request closed without merging**